dps/utils/Unfolding.py

```diff
@@ -88,6 +88,53 @@ def refold(self):
         else:
             print("Data has not been unfolded. You cannot refold something that hasn't been first unfolded")
         return self.refolded_data
+    def get_covariance_matrix(self):
+        '''
+        Get the covariance matrix from all contributions
+        https://root.cern.ch/doc/master/classTUnfoldDensity.html#a7f9335973b3c520e2a4311d2dd6f5579
+        '''
+        import uncertainties as u
+        from numpy import array, matrix, zeros
+        from numpy import sqrt as np_sqrt
+        if self.unfolded_data is not None:
+            # Calculate the covariance from TUnfold
+            covariance = asrootpy( 
+                self.unfoldObject.GetEmatrixTotal('Covariance'))
+            # Reformat into a numpy matrix
+            xs = list(covariance.x())
+            zs = list(covariance.z())
+            cov_matrix = matrix(zs)
+            bin_widths = array(xs)
+            corr = array(zeros((cov_matrix.shape[0], cov_matrix.shape[1]) ))
+            # Create a correlation matrix
+            for i in range(0,cov_matrix.shape[0]):
+                for j in range(0,cov_matrix.shape[1]):
+                    corr[i,j] = cov_matrix[i,j] / np_sqrt( cov_matrix[i,i] * cov_matrix[j,j] )
+                    
```

Comment:
The correlation matrix is not used anywhere at the moment. I can combine with the systematic correlations to create a similar total correlation matrix if it is useful